llvm/include/llvm/Support/GenericDomTreeConstruction.h

```diff
@@ -586,7 +586,8 @@ struct SemiNCAInfo {
     // Loop over all of the discovered blocks in the function...
     for (NodePtr W : llvm::drop_begin(NumToNode)) {
       // Don't replace this with 'count', the insertion side effect is important
-      if (DT.DomTreeNodes[W]) continue;  // Haven't calculated this node yet?
+      if (DT.getNode(W))
```

Comment:
The comment directly above this line claims that the insertion side effect is important. Either the new code or the old comment is wrong...

Reply:
WFIW, this refactor makes it less obvious whether `W` is inserted or not. If we do need the insertion, we should update the comment and make this explicit.

Otherwise +1 to what @nikic said above.

Reply:
Removed the comment, thanks for catching this.